internal/storage/insert_data.go

```diff
@@ -69,6 +73,14 @@ func NewInsertDataWithCap(schema *schemapb.CollectionSchema, cap int) (*InsertDa
 		if field.IsPartitionKey && field.GetNullable() {
 			return nil, merr.WrapErrParameterInvalidMsg("partition key field not support nullable")
 		}
+		if field.IsFunctionOutput {
+			if field.IsPrimaryKey || field.IsPartitionKey {
+				return nil, merr.WrapErrParameterInvalidMsg("function output field should not be primary key or partition key")
+			}
+			if !withFunctionOutput {
```

Comment:
Is it possible to perform an early check?

Reply:
can you clarify? what kind of checks are needed?

here we are just making a container object `*InsertData` based on the schema and haven't seen the actual data yet